tests/unit/jobs/test_models.py

```diff
@@ -78,3 +80,65 @@ def test__str__(self):
         self.assertEqual(
             "{}".format(self.submission.id), self.submission.__str__()
         )
+
+
+@pytest.mark.django_db
+class TestSubmissionModel:
+    def setup_method(self, method):
+        self.user = User.objects.create_user(
+            username="testuser", password="password"
+        )
+        self.challenge_host_team = ChallengeHostTeam.objects.create(
+            team_name="Test Challenge Host Team", created_by=self.user
+        )
+        self.challenge = Challenge.objects.create(
+            title="Test Challenge",
+            description="Description for test challenge",
+            terms_and_conditions="Terms and conditions for test challenge",
+            submission_guidelines="Submission guidelines for test challenge",
+            creator=self.challenge_host_team,
+            start_date=timezone.now() - timedelta(days=2),
+            end_date=timezone.now() + timedelta(days=1),
+            published=False,
+            enable_forum=True,
+            anonymous_leaderboard=False,
+        )
+        self.challenge_phase = ChallengePhase.objects.create(
+            name="Test Phase",
+            challenge=self.challenge,
+            max_submissions=5,
+            max_submissions_per_day=2,
+            max_submissions_per_month=10,
+        )
+        self.participant_team = ParticipantTeam.objects.create(
+            team_name="Test Participant Team", created_by=self.user
+        )
+
+    def test_max_submissions_per_day_reached(self):
```

Comment:
Hey @TheArchitect19 , While the test for "maximum submissions per day" is thorough, there could be additional tests for other constraints or scenarios:
Test for max_submissions (overall limit).
Test for max_submissions_per_month.